cluster-autoscaler/integration/integration_test.go

```diff
@@ -336,4 +333,64 @@ func (driver *Driver) controllerTests() {
 			})
 		})
 	})
+	Describe("testing scale down when priority of other machine is 1", func() {
+		Context("set priority of a node to 1 and trigger scale down of another node in the same node group", func() {
+			It("should update priority of other node to 3 and remove the correct machine", func() {
+				By("Deploying a workload")
+				Expect(driver.deployWorkload(int32(1), scaleUpWorkload, workerWithOneZone, false)).To(BeNil())
+				By("Validating Scale up")
+				Eventually(
+					driver.targetCluster.getNumberOfReadyNodes,
+					pollingTimeout,
+					pollingInterval).
+					Should(BeNumerically("==", initialNumberOfNodes+1))
+
+				By("Setting priority annotation on the newly scaled up machine")
+				mcdList, err := driver.controlCluster.MCMClient.MachineV1alpha1().MachineDeployments(controlClusterNamespace).List(context.TODO(), metav1.ListOptions{})
+				Expect(err).To(BeNil())
+
+				var mcdName string
+				for _, mcd := range mcdList.Items {
+					if strings.Contains(mcd.Name, workerWithOneZone) {
+						mcdName = mcd.Name
+						break
+					}
+				}
+
+				mcList, err := driver.controlCluster.MCMClient.MachineV1alpha1().Machines(controlClusterNamespace).List(context.TODO(), metav1.ListOptions{
+					LabelSelector: metav1.FormatLabelSelector(&metav1.LabelSelector{MatchLabels: map[string]string{mcdNameLabel: mcdName}})})
+				Expect(err).To(BeNil())
+
+				clone := mcList.Items[0].DeepCopy()
+				clone.Annotations[mcmPriorityAnnotation] = "1"
+				_, err = driver.controlCluster.MCMClient.MachineV1alpha1().Machines(controlClusterNamespace).Update(context.TODO(), clone, metav1.UpdateOptions{})
+				Expect(err).To(BeNil())
+
+				By("Deploying another workload")
+				Expect(driver.deployWorkload(int32(1), scaleUpWorkload+"-1", workerWithOneZone, false)).To(BeNil())
+				By("Validating Scale up")
+				Eventually(
+					driver.targetCluster.getNumberOfReadyNodes,
+					pollingTimeout,
+					pollingInterval).
+					Should(BeNumerically("==", initialNumberOfNodes+2))
+
+				By("Scaling down newer workload to zero...")
+				Expect(driver.scaleWorkload(scaleUpWorkload+"-1", 0)).To(BeNil())
```

Comment:
name the other one with "-A"
```suggestion
				Expect(driver.scaleWorkload(scaleUpWorkload+"-B", 0)).To(BeNil())
```